pkg/storage/s3/s3.go

```diff
@@ -236,27 +238,48 @@ func (d *driver) getS3Service() (*s3.S3, error) {
 		return nil, err
 	}
 
+	rootCAs, err := x509.SystemCertPool()
+	if err != nil {
+		return nil, fmt.Errorf("unable to load system root CA bundle: %w", err)
+	}
+
+	userCABundle, err := d.getCABundle()
+	if err != nil {
+		return nil, err
+	}
+	rootCAs.AppendCertsFromPEM([]byte(userCABundle))
+
+	tr := &http.Transport{
+		Proxy: func(req *http.Request) (*url.URL, error) {
+			return httpproxy.FromEnvironment().ProxyFunc()(req.URL)
+		},
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}).DialContext,
+		TLSClientConfig: &tls.Config{
+			RootCAs: rootCAs,
+		},
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+		ForceAttemptHTTP2:     true,
+	}
+
+	err = http2.ConfigureTransport(tr)
```

Comment:
We need to call this in the swift integration code as well[1].
[1] https://github.com/openshift/cluster-image-registry-operator/blob/master/pkg/storage/swift/swift.go?plain=1#L225